### ISSUE TITLE
CS8604 false positive

### ISSUE THREAD

**Issue by stepagrus** [dotnet/roslyn#81766]
**.NET SDK 10.0.101, **: 

**Steps to Reproduce**:

Below is a screenshot of a typical useful scenario where a false positive occurs.

<img width="968" height="566" alt="Image" src="https://github.com/user-attachments/assets/9f6dbcb4-25cb-4d4a-a279-4861a177952e" />

Here is the source code, which will compile without dependencies.
```
public class Class2
{
  public Class2()
  {
    Foo(Some);
  }

  public string Some => field ?? "some string";

  public static void Foo(string command)
  {
  }
}
```

**Diagnostic Id**:
`Possible null reference argument for parameter 'command' in 'void Class2.Foo(string command)'.`

**Expected Behavior**:
the warning should not occur

**Actual Behavior**:
False positive warning CS8604

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/81331
